text_extensions_for_pandas/io/conll.py

```diff
@@ -108,19 +124,35 @@ def token_metadata(self) -> Dict[str, List[str]]:
     def line_nums(self):
         return self._line_nums
 
-    def add_line(self, line_num: int, line_elems: List[str]):
-        """
-        :param line_num: Location in file, for error reporting
-        :param line_elems: Fields of a line, pre-split
-        """
-        if len(line_elems) != 1 + len(self._column_names) :
-            raise ValueError(f"Unexpected number of elements {len(line_elems)} "
-                             f"at line {line_num}; expected "
-                             f"{1 + len(self._column_names)} elements.")
-        token = line_elems[0]
-        raw_tags = line_elems[1:]
-        self._tokens.append(token)
-        self._line_nums.append(line_num)
+    @property
+    def doc_id(self):
+        return self._doc_id
+
+    @property
+    def paragraph_id(self):
+        return self._paragraph_id
+
+    @property
+    def sentence_id(self):
+        return self._sentence_id
+
+    @property
+    def column_names(self):
+        return self._column_names
+
+    @property
+    def has_conll_u_metadata(self):
+        return (self._sentence_id is not None) or (self._paragraph_id is not None) or (self._doc_id is not None)
+
+    def set_conll_u_metadata(self, doc_id: str = None, paragraph_id: str = None, sent_id: str = None):
```

Comment:
Would you mind augmenting these fields with a configurable set of key-value pairs? The CoNLL-U spec seems to think the comments before each sentence are supposed to hold arbitrary named attributes; see https://universaldependencies.org/format.html#sentence-boundaries-and-comments